### DISCUSSION TITLE
Rust version

### DISCUSSION THREAD

**Discussion by bcvanmeurs** [rapidfuzz/RapidFuzz#331] [General]
Hi Max,

Thanks for creating this awesome library. I am interested in porting and maintaining a rust version of RapidFuzz.
I see that you are an owner of this crate: https://docs.rs/rapidfuzz/latest/rapidfuzz/ but I can't seem to find a public repo for it.

Would you be up for letting me implement/extend the rust version? I prefer working with you on keeping it within this project over creating a fork.

Regards,
Bram

**Comment by bcvanmeurs** [comment]
I just found a reference to https://github.com/maxbachmann/rapidfuzz-rs in the issues but this seems to be removed/private.

**Reply by maxbachmann** [reply]
Hi Bram,

I would love to have a more complete + maintained rust implementation. So if you are up to implementing more of the port this would be awesome. I can help with any questions in regards to the algorithms and their C++ implementation.

I made my old initial implementation public again. I implemented this only to play around with Rust and basically needs a complete rewrite.

**Comment by maxbachmann** [comment]
I started working on a rewrite in rust in https://github.com/maxbachmann/rapidfuzz-rs/tree/rewrite.

**Comment by maxbachmann** [comment]
I published a first version of the rust port. It mostly includes the basic metrics. Still missing are:
- most of the fuzz module
- simd implementations for Many x Many comparisons of short strings
- editops / opcodes to find edit operations needed for a conversion

You can find it here: https://crates.io/crates/rapidfuzz. Let me know if you run into any issues or have suggestions for improvements for the API.
